qiskit_experiments/calibration_management/update_library.py

```diff
@@ -184,6 +184,62 @@ class Drag(BaseUpdater):
     __fit_parameter__ = "beta"
 
 
+class FineDrag(BaseUpdater):
+    """Updater for the fine drag calibration."""
+
+    # pylint: disable=arguments-differ,unused-argument
+    @classmethod
+    def update(
+        cls,
+        calibrations: Calibrations,
+        exp_data: ExperimentData,
+        parameter: str,
+        schedule: Union[ScheduleBlock, str],
+        result_index: Optional[int] = -1,
+        group: str = "default",
+        target_angle: float = np.pi,
+        **options,
+    ):
+        """Update the value of a drag parameter measured by the FineDrag experiment.
+
+        Args:
+            calibrations: The calibrations to update.
+            exp_data: The experiment data from which to update.
+            parameter: The name of the parameter in the calibrations to update.
+            schedule: The ScheduleBlock instance or the name of the instance to which the parameter
+                is attached.
+            result_index: The result index to use. By default search entry by name.
+            group: The calibrations group to update. Defaults to "default."
+            target_angle: The target rotation angle of the pulse.
+            options: Trailing options.
+
+        Raises:
+            CalibrationError: If we cannot get the pulse's standard deviation from the schedule.
+        """
+        qubits = exp_data.metadata["physical_qubits"]
+
+        if isinstance(schedule, str):
+            schedule = calibrations.get_schedule(schedule, qubits)
+
+        # Obtain sigma as it is needed for the fine DRAG update rule.
+        sigma = None
+        for block in schedule.blocks:
+            if isinstance(block, Play) and hasattr(block.pulse, "sigma"):
+                sigma = getattr(block.pulse, "sigma")
+
+        if sigma is None:
+            raise CalibrationError(f"Could not infer sigma from {schedule}.")
+
+        d_theta = BaseUpdater.get_value(exp_data, "d_theta", result_index)
+
+        d_delta = -0.25 * np.sqrt(np.pi) * d_theta * sigma / ((target_angle ** 2) / 4)
+
+        old_beta = calibrations.get_parameter_value(parameter, qubits, schedule, group=group)
```

Comment:
Can we assume `old_beta` is always in the calibrations?

Reply:
I think so yes.

Reply:
To run the calibration, you need to be able to get the schedule with the parameter, so it should be fine. An alternative would be to dig into the pulse like what is done above for sigma, but using `get_parameter_value` allows for a user to parametrize beta with a different name if they want to.

Reply:
That sounds reasonable. Parsing pulse schedule looks bit hackly.